server/routes/polls/polls.route.js

```diff
@@ -138,4 +139,102 @@ router.get(
   }
 );
 
+// @route GET /polls/:pollId
+// @desc To fetch a single poll using pollId
+// @params none
+// @access private
+router.get(
+  '/:pollId',
+  passport.authenticate('jwt', { session: true }),
+  async (req, res) => {
+    const poll = await Poll.findById(req.params.pollId);
+
+    if (!poll) {
+      return res.status(404).json({ message: 'Poll not found' });
+    }
+
+    return res.status(200).json(poll);
+  }
+);
+
+// @route POST /polls/:pollId/:imageIdx/vote
+// @desc To vote on poll
+// @params none
+// @access private
+router.post(
+  '/:pollId/:imageIdx/vote',
+  passport.authenticate('jwt', { session: true }),
+  async (req, res) => {
+    const imageIdx = req.params.imageIdx;
+    if (imageIdx !== 0 || imageIdx !== 1) {
+      return res.status(401).json({
+        message: 'You must include imageIdx in API URL. And it must be 0 or 1.',
+      });
+    }
+
+    const poll = await Poll.findById(req.params.pollId);
+
+    if (!poll) {
+      return res.status(404).json({ message: 'Poll not found' });
+    }
+
+    const vote = new Vote({
+      userId: req.user._id,
+      pollId: req.params.pollId,
+      pollImageIdx: imageIdx,
+    });
+    await vote.save();
+
+    if (!vote) {
+      return res.status(400).json({ message: 'Vote failed to save' });
+    }
+
+    poll.images[imageIdx].voteIds.push(vote._id);
+    await poll.save();
```

Comment:
Something to consider about this solution (I think this is fine), is we may want to make this change in a transaction (https://mongoosejs.com/docs/transactions.html) - this is a database concept where we want to make sure both of these actions happen together

Reply:
Would it be something like this?
```JavaScript
const mongoose = require('mongoose');
const db = mongoose.connection;

let session = null;
Poll.createCollection()
  .then(() => db.startSession())
  .then(_session => {
    session = _session;
    session.startTransaction();

    poll.images[imageIdx].voteIds.push(vote._id);
    return poll.save();
  })
  .then(doc => assert.ok(doc))
  .then(() => session.commitTransaction())
  .then(() => session.endSession());
```

Reply:
You want to put all database actions in the Transaction, so that would mean also the updating and creation of the vote. The reason is you want both the creation of a vote and the updating of a poll to be transactional (atomic)